presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayContainsAllFunction.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.common.block.Block;
+import com.facebook.presto.common.type.StandardTypes;
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.operator.aggregation.TypedSet;
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.ScalarFunction;
+import com.facebook.presto.spi.function.SqlNullable;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.spi.function.TypeParameter;
+
+@ScalarFunction("array_contains_all")
+@Description("Returns true if all elements of the second array are present in the first array")
+public final class ArrayContainsAllFunction
+{
+    private ArrayContainsAllFunction() {}
+
+    @TypeParameter("T")
+    @SqlType(StandardTypes.BOOLEAN)
+    @SqlNullable
+    public static Boolean containsAll(
+            @TypeParameter("T") Type elementType,
+            @SqlType("array(T)") Block firstArray,
+            @SqlType("array(T)") Block secondArray)
+    {
+        TypedSet firstSet = new TypedSet(elementType, firstArray.getPositionCount(), "arrayContainsAll");
```

Comment:
Can you benchmark this version vs. using OptimizedTypeSet and checking the cardinality of the intersection of the two sets?  I expect that using OptimizedTypedSet would be faster except for the case where a large second array is able to short circuit very early.

Reply:
Actually curions - how about CONTAINS? @rschlussel  did you fix that as well to be IS DISTINCT FROM? This function should behave like contains for that part

Reply:
contains fails for all null values. If we want this to fail for all nulls, we can do that as well, but we shouldn't have failures only for nulls nested in complex types, but otherwise compare nulls as equal (which is what typedset does by default)

Reply:
@jainavi17 did you have a chance to benchmark vs. a version using OptimizedTypeSet?